Sources/NIO/BaseSocketChannel.swift

```diff
@@ -798,7 +798,29 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
     // the `reset` event.
     final func reset() {
         self.readEOF()
-        self.close0(error: ChannelError.eof, mode: .all, promise: nil)
+
+        if self.lifecycleManager.isRegistered {
+            // if the socket is still open, let's try to get the actual socket error from the socket
```

Comment:
@weissi hmm this comment is confusing as we not really check for open but for registered above. Can you clarify ?

Reply:
good point. We close the socket when unregistering it but I should probably do something like this:

    if self.socket.isOpen {
        assert(self.lifecycleManager.isRegistered)
        ...

Reply:
fixed